server/repositories/TrustRepository.js

```diff
@@ -76,64 +76,84 @@ class TrustRepository extends BaseRepository {
-      )
-      .leftJoin(
-        'wallet as target_wallet',
-        'wallet_trust.target_wallet_id',
-        '=',
-        'target_wallet.id',
-      )
-      .where((builder) => this.whereBuilder(filter, builder))
-      .distinctOn('wallet_trust.id'); // distinct on id to avoid duplicates
+async getAllByFilter(filter, limitOptions) {
+  let subquery = this._session.getDB()
+    .select(
+      'wallet_trust.id',
+      'wallet_trust.actor_wallet_id',
+      'wallet_trust.target_wallet_id',
+      'wallet_trust.type',
+      'wallet_trust.originator_wallet_id',
+      'wallet_trust.request_type',
+      'wallet_trust.state',
+      'wallet_trust.created_at',
+      'wallet_trust.updated_at',
+      'wallet_trust.active',
+      'originator_wallet.name as originating_wallet',
+      'actor_wallet.name as actor_wallet',
+      'target_wallet.name as target_wallet',
+    )
+    .from('wallet_trust')
+    .leftJoin('wallet as originator_wallet', 'wallet_trust.originator_wallet_id', '=', 'originator_wallet.id')
+    .leftJoin('wallet as actor_wallet', 'wallet_trust.actor_wallet_id', '=', 'actor_wallet.id')
+    .leftJoin('wallet as target_wallet', 'wallet_trust.target_wallet_id', '=', 'target_wallet.id')
+    .where((builder) => this.whereBuilder(filter, builder))
+    .distinctOn('wallet_trust.id')
+    .orderBy('wallet_trust.id', 'asc'); 
 
-    // get the total count (before applying limit and offset options)
-    const count = await this._session.getDB().from(promise.as('p')).count('*');
+  let derivedTable = this._session.getDB().select('*').from(subquery.as('subquery'));
 
-    if (limitOptions && limitOptions.limit) {
-      promise = promise.limit(limitOptions.limit);
-    }
+  let order = 'desc';
+  let column = 'created_at';
 
-    if (limitOptions && limitOptions.offset) {
-      promise = promise.offset(limitOptions.offset);
+  if (limitOptions) {
+    if (limitOptions.order) {
+      order = limitOptions.order;
     }
+    if (limitOptions.sort_by) {
+      column = limitOptions.sort_by;
+    }
+  }
 
-    const result = await promise;
+  derivedTable = derivedTable.orderBy(column, order);
 
-    return { result, count: +count[0].count };
+  if (limitOptions && limitOptions.limit) {
+    derivedTable = derivedTable.limit(limitOptions.limit);
+  }
+  if (limitOptions && limitOptions.offset) {
+    derivedTable = derivedTable.offset(limitOptions.offset);
   }
+
+  const result = await derivedTable;
+
+  const countResult = await this._session.getDB()
+    .from('wallet_trust')
+    .leftJoin('wallet as originator_wallet', 'wallet_trust.originator_wallet_id', '=', 'originator_wallet.id')
+    .leftJoin('wallet as actor_wallet', 'wallet_trust.actor_wallet_id', '=', 'actor_wallet.id')
+    .leftJoin('wallet as target_wallet', 'wallet_trust.target_wallet_id', '=', 'target_wallet.id')
+    .where((builder) => this.whereBuilder(filter, builder))
+    .countDistinct('wallet_trust.id as count');
+
+  const count = countResult[0].count;
```

Comment:
I just thought that running a separate count query instead of extracting the count from the result variable will help to avoid potential inaccuracies that might arise from sorting or filtering operations on the result set. @Mloweedgar

Reply:
@samwel141